filebeat/tests/integration/autodiscover_test.go

```diff
@@ -0,0 +1,255 @@
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//This file was contributed to by generative AI
+
+//go:build integration && !requirefips
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/image"
+	"github.com/docker/docker/client"
+	"github.com/gofrs/uuid/v5"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/kind/pkg/apis/config/v1alpha4"
+	"sigs.k8s.io/kind/pkg/cluster"
+
+	"github.com/elastic/beats/v7/libbeat/tests/integration"
+	"github.com/elastic/elastic-agent-autodiscover/docker"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+func TestHintsDocker(t *testing.T) {
+	containerID := startFlogDocker(t)
+	filebeat := integration.NewBeat(
+		t,
+		"filebeat",
+		"../../filebeat.test",
+	)
+
+	cfgYAML := getConfig(t, nil, "autodiscover", "docker.yml")
+	filebeat.WriteConfigFile(cfgYAML)
+	filebeat.Start()
+
+	// By ensuring the Filestream input started with the correct ID, we're
+	// testing that the whole autodiscover + hints is working as expected.
+	filebeat.WaitForLogs(
+		fmt.Sprintf(
+			`"message":"Input 'filestream' starting","service.name":"filebeat","id":"container-logs-%s"`,
+			containerID,
+		),
+		30*time.Second,
+		"Filestream did not start for the test container")
+}
+
+func TestHintsKubernetes(t *testing.T) {
+	filebeat := integration.NewBeat(
+		t,
+		"filebeat",
+		"../../filebeat.test",
+	)
+
+	kubeConfigPath, noneName, containerID := startFlogKubernetes(t, filebeat.TempDir())
+
+	cfgYAML := getConfig(
+		t,
+		map[string]any{
+			"kubeConfig": kubeConfigPath,
+			"nodeName":   noneName,
+		},
+		"autodiscover",
+		"k8s.yml")
+	filebeat.WriteConfigFile(cfgYAML)
+	filebeat.Start()
+
+	// By ensuring the Filestream input started with the correct ID, we're
+	// testing that the whole autodiscover + hints is working as expected.
+	filebeat.WaitForLogs(
+		fmt.Sprintf(
+			`"message":"Input 'filestream' starting","service.name":"filebeat","id":"container-logs-%s"`,
+			containerID,
+		),
+		30*time.Second,
+		"Filestream did not start for the test container")
+}
+
+func startFlogKubernetes(t *testing.T, tempDir string) (string, string, string) {
+	uid := uuid.Must(uuid.NewV4()).String()
+
+	defer func() {
+		if t.Failed() {
+			t.Log("To see the Kind logs search for 'cluster.ProviderWithLogger' and uncomment it.")
+		}
+	}()
+	provider := cluster.NewProvider(
+	// Uncomment the next line to have Kind logs written to stderr.
+	// You will also have to import "sigs.k8s.io/kind/pkg/cmd"
+	// cluster.ProviderWithLogger(cmd.NewLogger()),
```

Comment:
Can't we have this enabled on `testing.Verbose()`?

Reply:
I could, but I don't like how it pollutes the output on CI. We always run tests with the `-v` flag.